d3a_api_client/rest_device.py

```diff
@@ -166,9 +166,20 @@ def _on_tick(self, message):
         self.callback_thread.submit(execute_function_util, function=function,
                                     function_name="on_tick")
 
-    def _on_trade(self, message):
-        logging.info(f"<-- {message.get('buyer')} BOUGHT {round(message.get('energy'),4)} kWh "
+    @staticmethod
+    def _log_trade_info(message):
+        logging.info(f"<-- {message.get('buyer')} BOUGHT {round(message.get('energy'), 4)} kWh "
                      f"at {round(message.get('price'), 2)}/kWh -->")
+
+    def _on_trade(self, message):
+        if "context" not in message:
+            # Device message
+            self._log_trade_info(message)
+        else:
+            # Aggregator message
+            for individual_trade in message["content"]:
```

Comment:
```suggestion
            for "individual_trade" in message["context"]:
```
Not 100% sure though

Reply:
Not really. The message["content"] in the aggregator message contains a list of dicts.